### PR TITLE
Inject esplora provider into core functions

### PR DESCRIPTION
**Pull request by victorkirov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Enhancement

# 📜 Background
We allow the user to change the base Electrs URL that their wallet uses (defaults to mempool.space/api). This was ignored in a variety of core functions which created their own esplora provider instance without considering this URL.

A change was made in Core to have the esplora provider injected into the functions instead of them making the instance themselves. This change required all calls to those functions to inject the esplora provider in the client code, allowing the custom URL to be used.

This will allow people with a lot of UTXOs in their payment address to change the electrs provider and be able to spend their funds.

# 🔄 Changes
The provider is now injected where needed. Before this, changing the URL in settings would still result in the use of mempool.space. This will actualyl change the URL used.

Impact:
The following pages were changed to inject the provider:
- send Btc txn (software and ledger)
- send ordinal (software and ledger) 
- edit fees
- create inscription
- restore ordinal
- send BRC20
- send rare sats
- sign PSBT  (software and ledger)

core pr: https://github.com/secretkeylabs/xverse-core/pull/305

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [x] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [x] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/screens/signPsbtRequest/index.tsx [hunk 1]
```diff
@@ -461,7 +461,7 @@ function SignPsbtRequest() {
             titleFailed={signatureRequestTranslate('LEDGER.CONFIRM.ERROR_TITLE')}
             textFailed={signatureRequestTranslate('LEDGER.CONFIRM.ERROR_SUBTITLE')}
             imageDefault={ledgerConnectDefaultIcon}
-            isConnectSuccess={isTxApproved}
+            isConnectSuccess={false}
```
Comment (teebszet) [inline src/app/screens/signPsbtRequest/index.tsx:464]: was this intentional?

**Review by teebszet — COMMENTED** [review]
Yay for build time type checking!

**Review by teebszet — DISMISSED** [review]

File: src/app/utils/helper.ts [hunk 2]
```diff
@@ -132,30 +132,52 @@ export function checkNftExists(
...
+  const defaultBtcClient = new BitcoinEsploraApiProvider({
+    network,
+  });
+
+  try {
+    const [customHash, defaultHash] = await Promise.all([
+      btcClient.getBlockHash(1),
+      defaultBtcClient.getBlockHash(1),
+    ]);
+    // this ensures the URL is for correct network
```
Comment (teebszet) [inline src/app/utils/helper.ts:174]: 👏

File: src/app/hooks/useBtcClient.ts [hunk 3]
```diff
@@ -1,40 +1,21 @@
 import { BitcoinEsploraApiProvider } from '@secretkeylabs/xverse-core';
-import { useEffect, useMemo } from 'react';
+import { useMemo } from 'react';
 import useWalletSelector from './useWalletSelector';
 
 const useBtcClient = () => {
   const { network, btcApiUrl } = useWalletSelector();
   const { type, btcApiUrl: remoteBtcApiURL } = network;
+  const url = btcApiUrl || remoteBtcApiURL;
```
Comment (teebszet) [inline src/app/hooks/useBtcClient.ts:8]: what's the difference here between network.btcApiUrl and btcApiUrl

Comment (victorkirov) [reply src/app/hooks/useBtcClient.ts:8]: The one from wallet selector is the URL if they already changed it to a custom value. The one on the network object is the default URL.

File: src/app/screens/settings/changeNetwork/index.tsx [hunk 4]
```diff
@@ -88,12 +88,11 @@ function ChangeNetworkScreen() {
   const { t } = useTranslation('translation', { keyPrefix: 'SETTING_SCREEN' });
   const { network, btcApiUrl, networkAddress } = useWalletSelector();
   const [changedNetwork, setChangedNetwork] = useState<SettingsNetwork>(network);
-  const [error, setError] = useState<string>('');
+  const [stacksUrlError, setStacksUrlError] = useState<string>('');
   const [btcURLError, setBtcURLError] = useState('');
   const [btcUrl, setBtcUrl] = useState(btcApiUrl || network.btcApiUrl);
-  const [url, setUrl] = useState<string>(networkAddress || network.address);
+  const [stacksUrl, setStacksUrl] = useState<string>(networkAddress || network.address);
```
[the remark this answers is not in the document]
Comment (victorkirov) [reply src/app/screens/settings/changeNetwork/index.tsx:94]: Same as the bitcoin thing

**Pull request merged**